api/src/main/java/org/openmrs/module/addresshierarchy/config/AddressConfigurationLoader.java

```diff
@@ -0,0 +1,274 @@
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.GlobalProperty;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.addresshierarchy.AddressHierarchyLevel;
+import org.openmrs.module.addresshierarchy.service.AddressHierarchyService;
+import org.openmrs.module.addresshierarchy.util.AddressHierarchyImportUtil;
+import org.openmrs.util.OpenmrsConstants;
+import org.openmrs.util.OpenmrsUtil;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.io.xml.DomDriver;
+
+/**
+ * Responsible for loading the address configuration appropriately
+ */
+public class AddressConfigurationLoader {
+
+	protected static final String ADDR_CONFIG_FILE_NAME = "addressConfiguration.xml";
+
+	public static final String NOT_COMPUTABLE_CHECKSUM = "not_computable_checksum";
+	public static final String NOT_READABLE_CHECKSUM = "not_readadble_checksum";
+
+	public static final String CHECKSUM_FILE_EXT = "checksum";
+
+	protected static Log log = LogFactory.getLog(AddressConfigurationLoader.class);
+
+	/**
+	 * @return The path to the configuration subdirectory.
+	 */
+	public static String getSubdirConfigPath() {
+		return new StringBuilder()
+				.append(OpenmrsUtil.getApplicationDataDirectory())
+				.append("configuration").append(File.separator).append("addresshierarchy")
+				.toString();
+	}
+
+	public static void loadAddressConfiguration() {
+
+		final ConfigLoaderUtil configUtil = new ConfigLoaderUtil(getSubdirConfigPath());
+
+		String xmlConfigFileName = ADDR_CONFIG_FILE_NAME;
+
+		File configFile = configUtil.getFile(xmlConfigFileName);
+		if (!configFile.exists()) {
+			log.warn("Address hierarchy configuration file appears invalid, skipping the loading process: " + xmlConfigFileName);
+			return;
+		}
+
+		String lastChecksum = "";
+		String checksum = "";
+
+		//
+		// Processing the XML configuraton file
+		//
+		AddressConfiguration addressConfiguration = readFromFile(configFile);
+
+		boolean forceReloadEntries = false;
+		lastChecksum = configUtil.readLatestChecksum(xmlConfigFileName);
+		checksum = configUtil.computeChecksum(xmlConfigFileName);
+
+		if (checksum.equals(lastChecksum)) {
+			log.info("Address hierarchy configuration file is unchanged, skipping it: " + xmlConfigFileName);
+		}
+		else {
+			
+			log.info("Address hierarchy configuration file has changed, reloading it: " + xmlConfigFileName);
+			
+			if (!isMatchableLevelConfig(addressConfiguration.getAddressComponents()) && !addressConfiguration.mustWipe()) {
+				log.warn("The address hierarchy configuration was not loaded because of a mismatch between the exisiting and provided address hierarchy levels.");
```

Comment:
This is also a case where I would wonder whether this should be a fatal error, or we want the module to load up as normal.  I think a case can be made where the system should throw a hard-error if the expected configuration is not loaded as desired.